### PR TITLE
Deprecate TableRegistry.

### PR DESCRIPTION
**Pull request by ADmad**
Refs cakephp/cakephp#14666

### CODE REVIEW

File: en/core-libraries/events.rst [hunk 1]
```diff
@@ -229,11 +229,11 @@ a more direct approach and only listen to the event you really need::
 
     // You can create the following before the
     // save operation, ie. config/bootstrap.php
-    use Cake\ORM\TableRegistry;
+    use Cake\Datasource\FactoryLocator;
     // If sending emails
     use Cake\Mailer\Email;
 
-    TableRegistry::getTableLocator()->get('ThirdPartyPlugin.Feedbacks')
+    FactoryLocator::get('Table')->get('ThirdPartyPlugin.Feedbacks')
```
Comment (markstory) [inline en/core-libraries/events.rst:236]: Are you thinking that we are early enough in 4.x that we don't need the 'before x do y' type language?

Comment (dereuromark) [reply en/core-libraries/events.rst:236]: I personally liked the method call, instead of the additional magic string.

Comment (ADmad) [reply en/core-libraries/events.rst:236]: @markstory Usage like `FactoryLocator::get('Table')->get('ThirdPartyPlugin.Feedbacks')` work even in 3.x. So we don't need any `before x do y` type language.

Comment (othercorey) [reply en/core-libraries/events.rst:236]: I believe he means `"Prior to 4.1, use ...`

Comment (ADmad) [reply en/core-libraries/events.rst:236]: @othercorey Yes I get that, but it's not necessary since the new usage shown actually works since 3.x.

Comment (dereuromark) [reply en/core-libraries/events.rst:236]: I still think those should be kept, as the `get('Table')` part should not be leaking to the developer, mainly for usability.

Comment (ADmad) [reply en/core-libraries/events.rst:236]: Practically everywhere else the examples have been updated to use `$this->getTableLocator()`. This particular example shows setting an event listener in `bootstrap.php` for global event manager using a closure, hence using `FactoryLocator` directly is the only way.

**Pull request merged**